app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,120 @@
+//Todo
+// 5. build. gradel Tpfvm flqb
+
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+
+  private Long id = 0L;
+  private HashMap<Long, Task> tasks = new HashMap();
+
+  /**
+   * http get 요청을 받아 알일들을 리턴합니다.
+   *
+   * @return tasks를 list로 반환
+   */
+  @GetMapping
+  @ResponseStatus(HttpStatus.OK)
+  public List<Task> taskList() {
+    return tasks
+        .values()
+        .stream()
+        .collect(Collectors.toList());
+  }
+  /**
+   * http get 요청을 id와 함꼐 받아서 그 해당 id에 해당하는 할일을 반환합니다.
+   *
+   * @param id
+   * @return 해당 id task
+   * @throws HttpClientErrorException task가 없는 id로 요청시
+   */
```

Comment:
```suggestion
  /**
   * 주어진 id에 해당하는 할 일을 검색해 리턴합니다.
   *
   * @param id 할 일의 아이디
   * @return 검색 결과 찾아낸 할 일
   * @throws TaskNotFoundException 할 일을 찾지 못한 경우
   */
```

예외가 `HttpClientErrorException`인 것은 아쉬운 일입니다. 주석에 적은 설명에 잘 어울리는 예외 클래스를 사용해 줍시다.

`@throws`는 `~한 경우`로 표현을 통일시켜 줍시다. `if`의 의미를 살려주기 위해서에요.

다음은 `String` 클래스의 생성자입니다. `@throws` 부분을 잘 읽어보세요.

![image](https://user-images.githubusercontent.com/1855714/129900174-62ee4099-4548-42f8-b8c4-31720d21c290.png)